src/soundio/sounddeviceportaudio.cpp

```diff
@@ -280,17 +280,33 @@ SoundDeviceError SoundDevicePortAudio::open(bool isClkRefDevice, int syncBuffers
             // Clear first 1.5 chunks on for the required artificial delaly to
             // a allow jitter and a half, because we can't predict which
             // callback fires first.
-            m_outputFifo->releaseWriteRegions(
-                    m_outputParams.channelCount * m_framesPerBuffer * kFifoSize
-                            / 2);
+            int writeCount = m_outputParams.channelCount * m_framesPerBuffer *
+                    kFifoSize / 2;
+            CSAMPLE* dataPtr1;
+            ring_buffer_size_t size1;
+            CSAMPLE* dataPtr2;
+            ring_buffer_size_t size2;
+            (void)m_outputFifo->aquireWriteRegions(writeCount, &dataPtr1,
+                    &size1, &dataPtr2, &size2);
+            memset(dataPtr1, 0, sizeof(CSAMPLE) * size1);
```

Comment:
I haven't checked if it would make sense to introduce helper functions to reduce the redundant code when dealing with FIFO. At some point even a derived or specialized `SampleFifo` would make sense. Just some ideas, we don't need to fix this now.